packages/core/src/RushTool.ts

```diff
@@ -0,0 +1,64 @@
+import path from "path";
+import fs from "fs-extra";
+import jju from "jju";
+
+import { PackageJSON } from "@changesets/types";
+import { Tool, ToolType, Package, Packages, MonorepoRoot } from './Tool';
+
+interface RushJson {
+    projects: RushProject[];
+}
+
+interface RushProject {
+    packageName: string;
+    projectFolder: string;
+}
+
+export const RushTool: Tool = {
+    type(): ToolType {
+        return 'rush';
+    },
+
+    async isMonorepoRoot(directory: string): Promise<MonorepoRoot | undefined> {
+        try {
+            await fs.readFile(path.join(directory, "rush.json"), "utf8");
+            return {
+                dir: directory,
+                tool: RushTool
+            };
+        } catch (err) {
+            if (err.code !== "ENOENT") {
+                throw err;
+            }
+        }
+    },
+
+    async getPackages(directory: string): Promise<Packages | undefined> {
+        try {
+            const rushText: string = await fs.readFile(path.join(directory, "rush.json"), "utf8");
+
+            // Rush configuration files are full of inline and block-scope comment blocks (JSONC),
+            // so we use a parser that can handle that.
+            const rushJson: RushJson = jju.parse(rushText);
```

Comment:
TIL about `jju` - nice 👍  I was using recently `golden-fleece` to manipulate JSON5 and I had some formatting issues with it. Definitely gonna give this a try next time.